### ISSUE TITLE
Something rustles nearby! & Pathing

### ISSUE THREAD

**Issue by Drambuie** [PokemonGoF/PokemonGo-Bot#2229]
### Expected Behavior

Bot should follow defined GPS path and catch Pokemon along the way and from Lure's at the stops it passes

### Actual Behavior

Bot only appears to capture Pokemon it directly walks over, 

Something rustles nearby! Pokemon are ignored

Lured pokemon at fort 15965893644d42479895bd99e6afa5de.16 are also ignored

Example : -

[14:45:36]  Now at Pokestop: Tower of London Yeoman Warders 
[14:45:36]  Spinning ... 
[14:45:37]  Loot:  
[14:45:37]  50 xp 
[14:45:37]  - 2x Pokeball (Total: 62) 
[14:45:37]  - 1x Revive (Total: 1) 
[14:45:37] PokeStop on cooldown. Time left: 5 minutes, 11 seconds
[14:45:43]  [x] Next egg incubates in 1.84 km 
[14:45:49]  -- Discarded 2x Pokeball (keeps only 60 maximum)  
[14:45:49]  -- Discarded 1x Revive (keeps only 0 maximum)  
[14:45:54] Lured pokemon at fort 0070a0b09ca14120a5f94b73a1e9b711.12
[14:46:05] Lured pokemon at fort 0070a0b09ca14120a5f94b73a1e9b711.12
[14:46:16] Something rustles nearby!
[14:46:23] Lured pokemon at fort 0070a0b09ca14120a5f94b73a1e9b711.12
[14:46:43] Something rustles nearby!
[14:46:48] Lured pokemon at fort 6ea9863f4e70437590ce7df239765550.16
[14:47:04] Something rustles nearby!
[14:47:09] Lured pokemon at fort 6ea9863f4e70437590ce7df239765550.16
[14:47:14]  Now at Pokestop: Royal Crown Jewels 
[14:47:14]  Spinning ... 
[14:47:15]  Loot:  
[14:47:15]  50 xp 
[14:47:15]  - 1x Pokeball (Total: 61) 
[14:47:15]  - 1x Greatball (Total: 101) 
[14:47:15]  - 1x Razz Berry (Total: 51) 
[14:47:15] PokeStop on cooldown. Time left: 5 minutes, 11 seconds

### Steps to Reproduce

Latest Dev build with following config

    "tasks": [
      {
        "type": "HandleSoftBan"
      },
      {
        "type": "CollectLevelUpReward"
      },
      {
        "type": "IncubateEggs",
        "config": {
          "longer_eggs_first": true
        }
      },
      {
        "type": "TransferPokemon"
      },
      {
        "type": "EvolveAll",
        "config": {
          "evolve_all": "Pidgey,Rattata,Weedle,Zubat",
          "evolve_cp_min": 300,
          "evolve_speed": 20,
          "use_lucky_egg": false
        }
      },
      {
        "type": "RecycleItems",
        "config": {
          "item_filter": {
          "Pokeball":       { "keep" : 60 },
          "Greatball":      { "keep" : 100 },
   	  "Ultraball":	    { "keep" : 150 },
          "Potion":         { "keep" : 0 },
          "Super Potion":   { "keep" : 0 },
          "Hyper Potion":   { "keep" : 0 },
          "Revive":         { "keep" : 0 },
          "Razz Berry":     { "keep" : 50 }
          }
        }
      },
      {
        "type": "CatchVisiblePokemon"
      },
      {
        "type": "CatchLuredPokemon"
      },
      {
        "type": "SpinFort"
      },
      {
        "type": "FollowPath",
        "config": {
          "path_mode": "loop",
          "path_file": "configs/path.json"
        }
      }
    ],
    "max_steps": 5,
    "forts": {
      "avoid_circles": true,
      "max_circle_size": 50
    },
    "websocket_server": true,
    "walk": 13.00,
    "action_wait_min": 1,
    "action_wait_max": 4,
    "debug": false,
    "test": false,
    "health_record": true,
    "location_cache": true,
    "distance_unit": "km",
    "reconnecting_timeout": 15,
    "evolve_captured": "Pidgey,Rattata,Weedle,Zubat",
    "catch_randomize_reticle_factor": 1.0,
    "catch_randomize_spin_factor": 1.0,
    "catch": {
      "any": {"catch_above_cp": 0, "catch_above_iv": 0, "logic": "or"},
      "// Example of always catching Rattata:": {},
      "// Rattata": { "always_catch" : true }
    },
    "release": {
      "Lapras": { "always_release" : false },
      "Moltres": { "always_release" : false },
      "Zapdos": { "always_release" : false },
      "Articuno": { "always_release" : false },
      "Mewtwo": { "always_release" : false },
      "Dragonite": { "always_release" : false },
      "Snorlax": { "always_release" : false },
      "Mew": { "always_release" : false },
      "Arcanine": { "always_release" : false },
      "Vaporeon": { "always_release" : false },
      "Gyarados": { "always_release" : false },
      "Exeggutor": { "always_release" : false },
      "Muk": { "always_release" : false },
      "Weezing": { "always_release" : false },
      "Flareon": { "always_release" : false },
      "Growlithe": { "always_release" : false },
      "Dragonair": { "always_release" : false },
      "Grimer": { "always_release" : false },
      "Exeggcute": { "always_release" : false },
      "Eevee": { "always_release" : false },
      "Slowbro": { "always_release" : false },
      "Victreebel": { "always_release" : false },
      "Machamp": { "always_release" : false },
      "Poliwrath": { "always_release" : false },
      "Clefable": { "always_release" : false },
      "Nidoking": { "always_release" : false },
      "Venusaur": { "always_release" : false },
      "Charizard": { "always_release" : false },
      "Golduck": { "always_release" : false },
      "Nidoqueen": { "always_release" : false },
      "Vileplume": { "always_release" : false },
      "Blastoise": { "always_release" : false },
      "Omastar":  { "always_release" : false },
      "Aerodactyl": { "always_release" : false },
      "Golem": { "always_release" : false },
      "Wigglytuff": { "always_release" : false },
      "Dewgong": { "always_release" : false },
      "Ninetales": { "always_release" : false },
      "Magmar": { "always_release" : false },
      "Kabutops": { "always_release" : false },
      "Electabuzz": { "always_release" : false },
      "Starmie": { "always_release" : false },
      "Jolteon": { "always_release" : false },
      "Rapidash": { "always_release" : false },
      "Pinsir": { "always_release" : false },
      "Scyther": { "always_release" : false },
      "Tentacruel": { "always_release" : false },
      "Gengar": { "always_release" : false },
      "Hypno": { "always_release" : false },
      "Pidgeot": { "always_release" : false },
      "Rhydon": { "always_release" : false },
      "Seaking": { "always_release" : false },
      "Kangaskhan": { "always_release" : false },
      "Koffing": { "always_release" : false },
      "any": { "keep_best_cp" : 3 },
      "// Example of always releasing Rattata:": {},
      "// Rattata": { "always_release" : true },
      "// Example of keeping stronger (based on CP) Pidgey:": {},
      "// Pidgey": { "keep_best_cp" : true },
      "// Example of keeping stronger (based on IV) Zubat:": {},
      "// Zubat": { "keep_best_iv" : true }
    },
    "vips" : {
         "Any pokemon put here directly force to use Berry & Best Ball to capture, to secure the capture rate!": {},
        "any": {"catch_above_cp": 1200, "catch_above_iv": 0.9, "logic": "or" },
        "Lapras": {},
        "Moltres": {},
        "Zapdos": {},
        "Articuno": {},
        "Charmander": {},
	"Bulbasaur": {},
	"Pikachu": {},

        "// S-Tier pokemons (if pokemon can be evolved into tier, list the representative)": {},
        "Mewtwo": {},
        "Dragonite": {},
        "Snorlax": {},
        "// Mew evolves to Mewtwo": {},
        "Mew": {},
        "Arcanine": {},
        "Vaporeon": {},
        "Gyarados": {},
        "Exeggutor": {},
        "Muk": {},
        "Weezing": {},
        "Flareon": {}

    }
}

### Other Information
OS: 
Git Commit: (run 'git log -n 1 --pretty=format:"%H"' and paste it here)
Python Version: (run 'python -V' and paste it here)

**Comment by braingineer** [comment]
There needs to be some attention in the code to spawning pokemon; currently there isn't any that I've seen on dev branch.  

I've made some of the following location edits to have a behavior for `MoveToPokemon`

https://gist.github.com/braingineer/efae8620b88ec76ee58960afc0b72f8a

in the edits, I made it so that once a task starts doing stuff, it gets cached and called immediately.  there was a ton of overhead otherwise.  so you don't have to keep doing that.  

but one thing that would be important is ordering so that `MoveToPokemon` gets called before `FollowPath` so that if a pokemon is near, it prioritizes it.
